### PR TITLE
EOS-20375 UI: Admin user should be able to reset password for other a…

### PR DESCRIPTION
**Pull request by azam93**
# Frontend

# Problem Statement
 - _Overview: JIRA number/GitHub Issue added to PR: [Y/N]: Y EOS-20367, EOS-20375, EOS-20368
 
## Design
 - For Bug describe the fix here. NA
 - For feature, post the link to the solution page. NA
 - Any northbound interface change and has been notified to other gatekeepers [Y/N]: N

## Coding
 - Coding conventions are followed and code is consistent. Yes/No? Yes
 - Confirm All CODACY errors are resolved. Yes/No? Yes

## Testing
 - _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No? Yes
 - _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No? No
 - _Confirm Testing was performed with installed RPM. Yes/No? Yes 
 - _Confirm all the ut/regression/smoke test has been performed [Y/N] Yes

## Checklist
 - _PR is self-reviewed? Yes/No? Yes
 - GitHub Issue is updated Yes
 - Jira is updated
     - Check if the description is clear and explained.
     - Check Acceptance Criterion is defined.
           -  All the tests performed should be mentioned before Resolving a JIRA.
           - Verification needs to be done before marked as Closed/Verified
           - Any interface change Yes/No? Yes
           - Change notified to other gatekeepers: Yes/No?
           - Code reviews done and addressed: Yes/No?
           - Codacy issues reviewed and addressed: Yes/No?
           - Deployment test : Done/Not?
           - UT/smoke test: Done/Not?
           - Jira number added to PR: Yes/No? Yes
           - Github merge done using UI: Yes/No? No
           - Side effects on other features - deployment/update/node-replacement
           - Changes to quick-start-guide/community impact
           - All dependent component code PR are raised or already merged Yes/No
           - All dependent code already merged in landing branch Yes/No

### CODE REVIEW

**Review by kupranay — CHANGES_REQUESTED** [review]
please update this with correct template

File: gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue [hunk 1]
```diff
@@ -852,13 +858,109 @@ export default class CortxUserSettingLocal extends Vue {
...
   public async mounted() {
     await this.getUserData();
   }
 
+  public deleteUserActionCB(event:any, data:any){
+    this.onDeleteConfirmation(data.id);
+  }
+  public deleteActionCondition(record: any) {
+    const vueInstance: any = this;
+    if (vueInstance.$hasAccessToCsm(userPermissions.users + userPermissions.delete)) {
```
Comment (shrimetta) [inline gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue:872]: You can directly return vueInstance.$hasAccessToCsm(userPermissions.users + userPermissions.delete)

File: gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue [hunk 2]
```diff
@@ -852,13 +858,109 @@ export default class CortxUserSettingLocal extends Vue {
...
+      return true;
+    }
+  }
+  
+  public editUserActionCB(event:any, data:any){
+    this.onEditBtnClick(data)
```
Comment (shrimetta) [inline gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue:878]: semicolon is missing

File: gui/src/models/download.ts [hunk 3]
```diff
@@ -25,3 +25,10 @@ export interface AuditLogQueryParam {
     offset?: number;
     limit?: number;
 }
+
+export interface csmUserQueryParam {
```
Comment (shrimetta) [inline gui/src/models/download.ts:29]: Name of the interface should be PascalCase

File: gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue [hunk 4]
```diff
@@ -852,13 +858,109 @@ export default class CortxUserSettingLocal extends Vue {
...
+  }
+  public editActionCondition(record: any) {
```
Comment (shrimetta) [inline gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue:880]: Please refactor this method.

File: gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue [hunk 5]
```diff
@@ -1007,6 +1109,43 @@ export default class CortxUserSettingLocal extends Vue {
     return isAdmin;
   }
 
+  private isLoggedInUserMonitor() {
```
Comment (shrimetta) [inline gui/src/components/onboarding/system-config/user-settings/user-setting-local.vue:1112]: Can have one method instead of three. Please refactor.

**Review by shrimetta — CHANGES_REQUESTED** [review]
Please implement the review comments.

**Comment by kupranay** [comment]
[12:16 PM] Pranay Kumar
    Alexander Voronov i can see on the above build
Admin user is getting create but the URL fired is 
/api/v2/preboarding/user
​[12:16 PM] Pranay Kumar
    is this expected?
​[12:16 PM] Pranay Kumar
    or should it be using the same URL which was being used for CSM user creation on normal page?
​[12:17 PM] Alexander Voronov
    no, this is not expected at all
​[12:18 PM] Alexander Voronov
    

or should it be using the same URL which was being used for CSM user creation on normal page?

yes
​[12:18 PM] Pranay Kumar
    please update swagger document with new expectation... if not done already
Mughala Azam please update it on GUI

**Comment by azam93** [comment]
Current PR - https://github.com/Seagate/cortx-management-portal/pull/412

**Pull request closed without merging**